app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntryUpdate.kt

```diff
@@ -63,6 +64,14 @@ class BloodPressureEntryUpdate(
 
       is BackPressed -> if (model.activeScreen == BP_ENTRY) {
         dispatch(Dismiss as BloodPressureEntryEffect)
+      } else if (model.activeScreen == DATE_ENTRY) {
+        val dateText = formatToPaddedDate(model.day, model.month, model.twoDigitYear, model.year)
```

Comment:
Just a suggestion: Since the `BloodPressureEntryUpdate` already has access to the current calendar date via `dateInUserTimeZone`, we can remove the `year` property from the model and read it from `dateInUserTimeZone`.

I checked the property and it is only used in formatting the padded date.

Reply:
I want to rethink how validation is done. I am not comfortable passing in the validator to the Update function itself (which means we should guarantee that the validators are pure and it is easy for other developers to pick up this pattern without knowing the intention). Now that we have a different view of the solution. I think, this can be a part of the `formatToPaddedDate` refactoring update.